src/check_datapackage/internals.py

```diff
@@ -76,46 +76,103 @@ def _validation_errors_to_issues(
 ) -> list[Issue]:
     """Transforms `jsonschema.ValidationError`s to more compact `Issue`s.
 
-    The list of errors is:
-
-      - flattened
-      - filtered for summary-type errors
-      - filtered for duplicates
-      - sorted by error location
-
     Args:
         validation_errors: The `jsonschema.ValidationError`s to transform.
 
     Returns:
         A list of `Issue`s.
     """
-    issues = [
+    return sorted(
+        set(chain.from_iterable(map(_validation_error_to_issues, validation_errors)))
+    )
+
+
+def _validation_error_to_issues(error: ValidationError) -> list[Issue]:
```

Comment:
If these are only used in one location, they can stay in that file. I think how you had it previous was fine ☺️